### PR TITLE
Use catalog that matches users level

### LINKED ISSUE #14: Build workflow without logging in
Then download and run locally or upload later/elsewhere

### PR DESCRIPTION
**Pull request by sverhoeven**
Also loads catalog from disk instead of fetching it from GitHub each time.
Also switched from storing bartender access token in cookie to storing it in a session file with cookie with just session id.

Refs #21 
Refs #15
Fixes #14

This PR requires `main` branch of https://github.com/i-VRESSE/bartender with https://github.com/i-VRESSE/bartender/pull/56 merged in.

Code in app/bartender-client/ is generated with `npm run generate-client` and should not be reviewed on code quality.
The app/catalogs/*,json have created by running `npn run catalogs`.

To test:
1. Configure bartender with roles as described at https://github.com/i-VRESSE/bartended-haddock3/blob/efa9e4da4f35fd086674c4806415571cc0e2d4dc/README.md#haddock3-application
    * Change haddock3 command in config.yaml to cat $config` to mock running of haddock3
1. Spinup 
    * postgresql docker container
    * fill with `alembic upgrade head`
    * bartender serve
    * npm run dev
 3. Regster account
 4. Make account super user with `bartender super <email>`
 5. Logout & login
    *  should see admin page in login menu
6. Goto http://localhost:3000/admin/users and give yourself the guru role.
    * Check on http://localhost:3000/profile that role is assigned
8. At http://localhost:3000/applications/builder check right catalog is loaded.
   * In global parameters and executation section the guru level has `self-contained`, expert has `max_cpus`.
9. Load a worklow (for example https://github.com/i-VRESSE/workflow-builder/blob/main/packages/haddock3_catalog/public/examples/docking-protein-ligand.zip) & submit
10. Job should complete.

When you are not logged in then http://localhost:3000/applications/builder allows download workflow, but not submit (#14).

### CODE REVIEW

File: app/routes/admin/users.tsx [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+  const { users, roles } = useLoaderData<typeof loader>();
+  const { submit, state } = useFetcher();
+  return (
+    <main>
+      <h1 className="my-6 text-3xl">User admin</h1>
+      <table className="table w-full">
+        <thead>
+          <tr>
+            <th>Email</th>
+            <th>Super user?</th>
```
Comment (Peter9192) [inline app/routes/admin/users.tsx:70]:
```suggestion
            <th>Super user</th>
```

File: app/routes/applications/upload.tsx [hunk 2]
```diff
@@ -5,17 +5,22 @@ import {
...
   const name = "haddock3";
   const app = await applicationByName(name);
-  const access_token = await getAccessToken(request);
-  if (access_token === undefined) {
+  const accessToken = await getAccessToken(request);
+  if (accessToken === undefined) {
     throw new Error("Unauthenticated");
   }
+  const level = await getLevel(accessToken)
+  if (level === '') {
```
Comment (Peter9192) [inline app/routes/applications/upload.tsx:21]: Wouldn't it be better to check explicitly if level is in allowed roles?

**Review by Peter9192 — APPROVED** [review]
Pair-reviewed :-) 
You already addressed or logged most of my comments.

File: app/routes/applications/upload.tsx (hunk 2 above)
Comment (sverhoeven) [reply app/routes/applications/upload.tsx]: check moved to own method isSubmitAllowed

**Pull request merged**